File: frontend/src/api/alertEvent.ts
import { http } from './http'

export interface AlertEventHandleLog {
  id: number
  eventId: number
  actionType: string
  actionTypeName?: string
  beforeStatus?: string
  afterStatus?: string
  operatorName?: string
  operatorPhone?: string
  actionComment?: string
  createdAt?: string
}

export interface AlertNotifyLog {
  id: number
  eventId?: number
  ruleId?: number
  channelId: number
  channelType: string
  receiverValue?: string
  notifyTitle?: string
  notifyContent?: string
  sendStatus: string
  providerMsgId?: string
  failureReason?: string
  sentAt?: string
  createdAt?: string
}

export interface AiSummary {
  what?: string
  impact?: string
  causes?: string[]
  actions?: string[]
  error?: string
}

export interface AlertEventItem {
  id: number
  eventNo: string
  incidentId?: number
  ruleId: number
  ruleName?: string
  objectId: number
  objectType: string
  objectName: string
  metricCode: string
  metricName: string
  alertLevel: string
  eventStatus: string
  currentValue?: string
  thresholdValue?: string
  eventTitle: string
  eventContent?: string
  eventReason?: string
  aiSummary?: string
  aiSummaryStatus?: string
  firstTriggeredAt?: string
  lastTriggeredAt?: string
  confirmedAt?: string
  recoveredAt?: string
  closedAt?: string
  createdAt?: string
  updatedAt?: string
  handleLogs?: AlertEventHandleLog[]
  notifyLogs?: AlertNotifyLog[]
}

export function listAlertEvents(query: {
  objectType?: string
  alertLevel?: string
  eventStatus?: string
  keyword?: string
  limit?: number
} = {}) {
  return http.get<AlertEventItem[]>('/alert-events', { params: query })
}

export function getAlertEvent(id: number) {
  return http.get<AlertEventItem>(`/alert-events/${id}`)
}

export function handleAlertEvent(payload: {
  eventId: number
  actionType: 'CONFIRM' | 'RECOVER' | 'CLOSE' | 'COMMENT'
  operatorName?: string
  operatorPhone?: string
  actionComment?: string
}) {
  return http.post<AlertEventItem>('/alert-events/action', payload)
}

export function createTestAlertEvent(payload: {
  ruleId: number
  objectId: number
  currentValue?: string
  eventReason?: string
}) {
  return http.post<AlertEventItem>('/alert-events/test', payload)
}

export function summarizeAlertEvent(id: number) {
  return http.post<void>(`/alert-events/${id}/summarize`)
}

export function listEventNotifyLogs(eventId: number) {
  return http.get<AlertNotifyLog[]>(`/alert-events/${eventId}/notify-logs`)
}

export function retryNotify(notifyLogId: number) {
  return http.post<AlertNotifyLog>('/alert-events/notify-logs/retry', { notifyLogId })
}

export function parseAiSummary(raw?: string): AiSummary | undefined {
  if (!raw) return undefined
  try {
    return JSON.parse(raw) as AiSummary
  } catch {
    return undefined
  }
}

File: backend/src/main/java/com/aiops/alert/entity/AlertEvent.java
package com.aiops.alert.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.time.LocalDateTime;
import lombok.Data;

@Data
@TableName("alert_event")
public class AlertEvent {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String eventNo;
    private Long incidentId;
    private Long ruleId;
    private Long objectId;
    private String objectType;
    private String objectName;
    private String metricCode;
    private String metricName;
    private String alertLevel;
    /** PENDING / CONFIRMED / RECOVERED / CLOSED */
    private String eventStatus;
    private String currentValue;
    private String thresholdValue;
    private String eventTitle;
    private String eventContent;
    private String eventReason;
    /** AI 摘要 JSON */
    private String aiSummary;
    /** AI 思考过程 (reasoning_content) */
    private String aiReasoning;
    /** PENDING / SUCCESS / FAILED */
    private String aiSummaryStatus;

    private LocalDateTime firstTriggeredAt;
    private LocalDateTime lastTriggeredAt;
    private LocalDateTime confirmedAt;
    private LocalDateTime recoveredAt;
    private LocalDateTime closedAt;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
